externallib.php

```diff
@@ -1319,4 +1318,72 @@ public static function toggle_discussion_subscription_returns() {
         );
     }
 
+    //mark single post as read
+    /**
+     * Returns description of method parameters
+     *
+     * @return external_function_parameters
+     * @since Moodle 3.0
+     */
+    public static function mark_single_post_read_parameters() {
+        return new external_function_parameters(
+            array(
+                'postid' => new external_value(PARAM_INT, 'Post ID'),
+            )
+        );
+    }
+
+    /**
+     * Marks a single post as read, post that the user is replying to
+     *
+     * @param int $postid the post id we are going to reply to
+     *
+     * @return array the forum discussion details
+     * @since Moodle 2.5
+     */
+    public static function mark_single_post_read($postid) {
+        global $DB, $CFG, $USER;
+        require_once($CFG->dirroot . "/mod/hsuforum/lib.php");
+        require_once($CFG->dirroot . "/mod/hsuforum/mobilelib.php");
+
+        $params = self::validate_parameters(self::mark_single_post_read_parameters(),
+            array(
+                'postid' => $postid
+            ));
+        $config = get_config('hsuforum');
+        $now = time();
+        $cutoffdate = $now - ($config->oldpostdays * 24 * 3600);
+
+        try {
+            if (!$DB->record_exists('hsuforum_read', array('userid' => $USER->id, 'postid' => $postid))) {
+                $sql = "INSERT INTO {hsuforum_read} (userid, postid, discussionid, forumid, firstread, lastread)
+
+                SELECT ?, p.id, p.discussion, d.forum, ?, ?
+                  FROM {hsuforum_posts} p
+                       JOIN {hsuforum_discussions} d ON d.id = p.discussion
+                 WHERE p.id = ? AND p.modified >= ?";
+                $DB->execute($sql, array($USER->id, $now, $now, $postid, $cutoffdate));
+            }
+        } catch (Exception $e) {
+            throw new coding_exception($e->getMessage());
```

Comment:
Why throw an exception here? Is there a reason you want to explicitly terminate execution and have a coding exception thrown?

Reply:
no reason for throwing will return just the message

Reply:
throw new coding_exception($e->getMessage()); Why throw the coding exception?